modules/external-api/src/main/java/org/opencastproject/external/endpoint/WorkflowDefinitionsEndpoint.java

```diff
@@ -0,0 +1,272 @@
+  }
+
+  @GET
+  @Path("/")
+  @RestQuery(name = "getworkflowdefinitions", description = "Returns a list of workflow definition.", returnDescription = "", restParameters = {
+          @RestParameter(name = "withoperations", description = "Whether the workflow operations should be included in the response", isRequired = false, type = BOOLEAN),
+          @RestParameter(name = "withconfigurationpanel", description = "Whether the workflow configuration panel should be included in the response", isRequired = false, type = BOOLEAN),
+          @RestParameter(name = "filter", description = "A comma seperated list of filters to limit the results with. A filter is the filter's name followed by a colon \":\" and then the value to filter with so it is the form <Filter Name>:<Value to Filter With>.", isRequired = false, type = STRING),
+          @RestParameter(name = "sort", description = "Sort the results based upon a list of comma seperated sorting criteria. In the comma seperated list each type of sorting is specified as a pair such as: <Sort Name>:ASC or <Sort Name>:DESC. Adding the suffix ASC or DESC sets the order as ascending or descending order and is mandatory.", isRequired = false, type = STRING),
+          @RestParameter(name = "limit", description = "The maximum number of results to return for a single request.", isRequired = false, type = INTEGER),
+          @RestParameter(name = "offset", description = "The index of the first result to return.", isRequired = false, type = INTEGER) }, reponses = {
+          @RestResponse(description = "A (potentially empty) list of workflow definitions is returned.", responseCode = HttpServletResponse.SC_OK),
+          @RestResponse(description = "The request is invalid or inconsistent.", responseCode = HttpServletResponse.SC_BAD_REQUEST) })
+  public Response getWorkflowDefinitions(@HeaderParam("Accept") String acceptHeader,
+          @QueryParam("withoperations") boolean withOperations,
+          @QueryParam("withconfigurationpanel") boolean withConfigurationPanel, @QueryParam("filter") String filter,
+          @QueryParam("sort") String sort, @QueryParam("offset") Integer offset, @QueryParam("limit") Integer limit) {
+    Stream<WorkflowDefinition> workflowDefinitions;
+    try {
+      workflowDefinitions = workflowService.listAvailableWorkflowDefinitions().stream();
+    } catch (WorkflowDatabaseException e) {
+      logger.error("The workflow service was not able to get the workflow definitions: {}", getStackTrace(e));
+      return ApiResponses.serverError("Could not retrieve workflow definitions, reason: '%s'", getMessage(e));
+    }
+
+    // Apply filter
+    if (StringUtils.isNotBlank(filter)) {
+      for (String f : filter.split(",")) {
+        int sepIdx = f.indexOf(':');
+        if (sepIdx < 0 || sepIdx == f.length() - 1) {
+          logger.info("No value for filter {} in filters list: {}", f, filter);
+          continue;
```

Comment:
Hmm... shouldn't we return a 400 bad request here?

Reply:
This is inline with the other APIs (e.g. [events](https://github.com/opencast/opencast/blob/15b367378cf89b236e116aeb7ad031f340cf70d7/modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java#L609-L612) and [series](https://github.com/opencast/opencast/blob/15b367378cf89b236e116aeb7ad031f340cf70d7/modules/external-api/src/main/java/org/opencastproject/external/endpoint/SeriesEndpoint.java#L199-L202)).